tests/unit-tests/test-class-course.php

```diff
@@ -440,21 +440,49 @@ public function testGetViewResultsLinkCourseResultsPage() {
 
 	/**
 	 * Test that the parameters are not added when hook function called out of the course archive page.
+	 *
+	 * @param bool $is_post_type_archive Returns true if the WP_Query is from an archive page.
+	 * @param bool $is_main_query Returns true if the WP_Query is the main query.
+	 * @param int $is_main_query_call_count Indicates the expectation for how many times $is_main_query will be called.
+	 * @param bool $set_current_screen_to_admin_panel Setting this to true will make the is_admin check return true, and otherwise it will return false.
+	 *
+	 * @dataProvider data_testCourseArchiveOrderSetOrderByDoesNotAddParametersOutsideOfArchivePage
 	 */
-	public function testCourseArchiveOrderSetOrderByDoesNotAddParametersOutsideOfArchivePage() {
+	public function testCourseArchiveOrderSetOrderByDoesNotAddParametersOutsideOfArchivePage( $is_post_type_archive, $is_main_query, $is_main_query_call_count, $set_current_screen_to_admin_panel ) {
 		$wp_query = $this->createMock( 'WP_Query' );
-		// Mock request is *not* for the course archive page.
+		// Returns true for the course archive page, but also returns true when the query is from 'wp-admin -> Sensei LMS -> Courses' or 'shortcode'
 		$wp_query->expects( $this->once() )
 			->method( 'is_post_type_archive' )
 			->with( 'course' )
-			->willReturn( false );
+			->willReturn( $is_post_type_archive );
 		// We dont expect any parameter to be set the query.
 		$wp_query->expects( $this->never() )
 			->method( 'set' );
-
+		// Returns 'true' for actual course archive page query, returns 'false' for queries generated by other sources, like shortcodes
```

Comment:
Just as a suggestion...
I like to follow [the AAA pattern](https://freecontent.manning.com/making-better-unit-tests-part-1-the-aaa-pattern/) in Unit Tests. 

---
A rough summary of the approach. AAA stands for "Arrange, Act, Assert":
- at the beginning of the test we prepare the test "environment" (create the test object, initialize and setup all the mocks and other supportive entities, etc.)
  `$a = new A();`
- then we perform the action you want to test
  `$actual = $a->m();`
- in the end, we compare the actual result with an expected value
  `self::assertSame($expected, $actual);`
---

Here, in the test, the *assertion* is: 
```
// We dont expect any parameter to be set the query.
$wp_query->expects( $this->never() )
	->method( 'set' );
```
And we can't put it after *acting*:
```
Sensei_Course::course_archive_set_order_by( $wp_query );
```
But, for the future reader, it might be a bit more clear what exactly we want to check if we place these two statements as closely as possible:
```
// We dont expect any parameter to be set the query.
$wp_query->expects( $this->never() )
	->method( 'set' );
Sensei_Course::course_archive_set_order_by( $wp_query );
```